src/main/webapp/app/entities/drug/drug.tsx

```diff
@@ -41,12 +41,14 @@ export const Drug = (props: IDrugProps) => {
   const getUniqueGenes = (associations: IAssociation[]) => {
     const biomarkers: DrugAssocBiomarkers = {};
     associations?.forEach(val => {
-      for (const alteration of val.alterations) {
+      for (const alteration of val?.alterations ?? []) {
         const geneName = getGeneNameFromAlteration(alteration);
```

Comment:
getGeneNameFromAlteration should not return undefined when alteration is available. If alteration genes list is empty, the currently method returns empty string. We should keep it that way.